### PR TITLE
feat: add create and delete endpoint for subsidies

### PR DESCRIPTION
**Pull request by emrosarioa**
* POST /api/v1/subsidies/
* DELETE /api/v1/subsidies/{uuid}/

ent-6998

### Description
Adds the create and delete endpoint for subsides, using drf-spectacular to document the intended use cases

### Testing instructions

Add some, if applicable

### Merge checklist
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

### CODE REVIEW

File: enterprise_subsidy/apps/api/v1/views/subsidy.py [hunk 1]
```diff
@@ -174,3 +197,78 @@ def can_redeem(self, request, uuid):  # pylint: disable=unused-argument
...
+                    create_serializer.data['default_starting_balance'],
+                    create_serializer.data['default_revenue_category'],
+                    create_serializer.data['default_internal_only'],
+                )
+                if created:
+                    return Response(SubsidySerializer(subsidy).data, status=status.HTTP_201_CREATED)
+                if subsidy:
+                    return Response(SubsidySerializer(subsidy).data, status=status.HTTP_200_OK)
+                else:
+                    assert False
```
Comment (iloveagent57) [inline enterprise_subsidy/apps/api/v1/views/subsidy.py:235]: You might also want to raise a 500 here, too - it'd be a truly "exceptional" condition if attempting to get or create returned nothing.

File: enterprise_subsidy/apps/api/v1/exceptions.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    """
+    status_code = 500
+    default_detail = 'encountered an unexpected error in subsidies service'
+    default_code = 'server_error'
+
+    def __init__(self, *args, **kwargs):
+        self.code = kwargs['code']
+        self.developer_message = kwargs.pop('developer_message', None)
+        self.user_message = kwargs.pop('user_message', None)
+        super().__init__(*args, **kwargs)
```
Comment (pwnage101) [inline enterprise_subsidy/apps/api/v1/exceptions.py:21]: As discussed, I think I'd be interested in a way to have this result in a JSON response body including all the custom attributes when raised within a view.

File: enterprise_subsidy/apps/api/v1/views/subsidy.py [hunk 3]
```diff
@@ -41,7 +48,13 @@ def __init__(self, can_redeem, content_price, unit, existing_transaction):  # py
 
 
 class SubsidyViewSet(
-    PermissionRequiredForListingMixin, mixins.RetrieveModelMixin, mixins.ListModelMixin, viewsets.GenericViewSet
+    PermissionRequiredForListingMixin,
+    mixins.CreateModelMixin,
+    mixins.DestroyModelMixin,
+    mixins.ListModelMixin,
+    mixins.RetrieveModelMixin,
+    mixins.UpdateModelMixin,
```
Comment (pwnage101) [inline enterprise_subsidy/apps/api/v1/views/subsidy.py:56]: If this specific PR doesn't implement update(), then maybe we should remove mixins.UpdateModelMixin for now.

File: enterprise_subsidy/apps/api/v1/exceptions.py (hunk 2 above)
Comment (iloveagent57) [reply enterprise_subsidy/apps/api/v1/exceptions.py:21]: yeah, me too!  I think we can push this change through and follow up with a small PR focused on that aspect of things.

File: enterprise_subsidy/apps/api/v1/views/subsidy.py [hunk 4]
```diff
@@ -174,3 +197,78 @@ def can_redeem(self, request, uuid):  # pylint: disable=unused-argument
...
+        tags=['subsidy'],
+        request=SubsidyCreationRequestSerializer,
+        responses={
+            201: SubsidySerializer,
+            200: SubsidySerializer,
+            403: exceptions.PermissionDenied,
+            400: exceptions.ValidationError,
+            500: exceptions.APIException,
+        },
+    )
```
Comment (iloveagent57) [inline enterprise_subsidy/apps/api/v1/views/subsidy.py:211]: Awesome, I like how this stuff came together on these endpoints.

**Review by iloveagent57 — APPROVED** [review]

File: enterprise_subsidy/apps/api/v1/exceptions.py (hunk 2 above)
Comment (emrosarioa) [reply enterprise_subsidy/apps/api/v1/exceptions.py:21]: Sounds like a plan, I'll follow up in a small PR!

**Review by pwnage101 — APPROVED** [review]

**Pull request merged**